### PR TITLE
[WIP] refactor(Core): replace ACE guard/mutex with standard C++

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
- Depends on: https://github.com/azerothcore/azerothcore-wotlk/pull/3743
- Partially based on: https://github.com/cmangos/mangos-wotlk/commit/8ae67274a754f821f0bea6a5f771eab3df764a57

## CHANGES PROPOSED:
-  Remove all ACE guards and mutexes in favour of standard C++

## ISSUES ADDRESSED:
- Part of https://github.com/azerothcore/azerothcore-wotlk/issues/99

## TESTS PERFORMED:
- None

## HOW TO TEST THE CHANGES:

- Not ready yet

## KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/common/Common.h [hunk 1]
```diff
@@ -194,7 +192,7 @@ typedef std::vector<std::string> StringVector;
 #define MAX_QUERY_LEN 32*1024
 
 #define ACORE_GUARD(MUTEX, LOCK) \
-  ACE_Guard< MUTEX > ACORE_GUARD_OBJECT (LOCK); \
+  std::lock_guard< MUTEX > ACORE_GUARD_OBJECT (LOCK); \
```
Comment (Viste) [inline src/common/Common.h:195]: std::lock_guard<std::mutex>

Comment (Viste) [reply src/common/Common.h:194]: As you probably understood, this macro is not really needed with std.

**Comment by Hacki95** [comment]
This PR isn't needed anymore right?

**Comment by FrancescoBorzi** [comment]
yes, closing. Thanks @Hacki95

**Pull request closed without merging**